File: Easy/0696-Count Binary Substrings/Linear Scan.py
class Solution:
    def countBinarySubstrings(self, s: str) -> int:
        prev, curr = 0, 0
        ret = 0

        for i in range(len(s)):
            curr += 1
            if i == len(s) - 1 or s[i] != s[i + 1]:
                ret += min(prev, curr)
                prev, curr = curr, 0

        return ret

File: Easy/0696-Count Binary Substrings/Linear Scan.rb
# @param {String} s
# @return {Integer}
def count_binary_substrings(s)
    prev, curr = 0, 0
    ret = 0

    for i in 0...s.length
        curr += 1
        if i == s.length - 1 or s[i] != s[i + 1]
            ret += [prev, curr].min
            prev, curr = curr, 0
        end
    end

    return ret
end
